### PR TITLE
docs: indexing locally and running tests

### PR DESCRIPTION
**Pull request by rabi-siddique**
The PR updates the README. Mainly the sections related to running the indexer locally and how tests work in this repo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,14 +111,49 @@ The fastest way to get support is by [searching our documentation](https://acade
...
+
+If you encounter issues with the indexer:
+
+- Check the logs for any errors or warnings that might indicate what is wrong. Use the command:
+
+  ```sh
+  docker logs <container_id>
+  ```
+
+  Replace <container_id> with the actual ID of your Docker container involved in indexing.
```
[the remark this answers is not in the document]
Comment (Muneeb147) [reply README.md:143]: May be we can add the command here to get container Id from container name:

`docker inspect --format="{{.Id}}" container_name`

And I think container name is `subquery-node` in our case?

File: README.md [hunk 2]
```diff
@@ -111,14 +111,49 @@ The fastest way to get support is by [searching our documentation](https://acade
...
-1. Start up an A3P instance:
+1. Start up an A3P instance
 
-```sh
-docker run -p 26657:26657 -p 1317:1317 -p 9090:9090 ghcr.io/agoric/agoric-3-proposals:latest
-```
+   ```sh
+   docker run -p 26657:26657 -p 1317:1317 -p 9090:9090 ghcr.io/agoric/agoric-3-proposals:latest
+   ```
+
```
Comment (Muneeb147) [inline README.md:119]: Why this step?
I see we have a3p as a separate service in docker compose and `yarn dev` should also spawn it.

Screenshot 2024-12-24 at 7 15 35 PM

Comment (turadg) [reply README.md:119]: I was wondering why `ci` was necessary. Your comment here is helpful. Please include it somewhere in the README or source code.

File: README.md [hunk 3]
```diff
@@ -111,14 +111,47 @@ The fastest way to get support is by [searching our documentation](https://acade
...
+   - Use the provided interface to write and execute your GraphQL queries.
+
+### Troubleshooting the Indexer
+
+If you encounter issues with the indexer:
+
+- Check the logs of indexer for any errors or warnings that might indicate what is wrong. Use the command:
+
+  ```sh
+  docker logs -f agoric-subql-subquery-node-1
```
Comment (Muneeb147) [inline README.md:140]: Does this container name remain consistent on each compse up?

Comment (Muneeb147) [reply README.md:140]: Was just curious about the repo name prefix and *-1 postfix

**Review by turadg — APPROVED** [review]

**Review by Muneeb147 — APPROVED** [review]

**Pull request merged**